### ISSUE TITLE
Uncaught Error: Cannot find module 'dialog'Error: Cannot find module 'dialog'    at Function.Module._resolveFilename (module.js:351:15)    at Function.Module._load (module.js:293:25)    at Module.require (module.js:380:17)    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/atom/browser/lib/rpc-server.js:128:79)    at EventEmitter.emit (events.js:119:17)    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/atom/browser/api/lib/web-contents.js:99:23)    at EventEmitter.emit (events.js:119:17)

### ISSUE THREAD

**Issue by SaxxonPike** [atom/tree-view#399]
[Enter steps to reproduce below:]

1. Right-click a file in the left column. (The file I used was green, not added to Git, and empty.)
2. Click Delete.

**Atom Version**: 0.186.0
**System**: Mac OS X 10.10.2
**Thrown From**: [tree-view](https://github.com/atom/tree-view) package, v0.164.0

### Stack Trace

Uncaught Error: Cannot find module 'dialog'
Error: Cannot find module 'dialog'
    at Function.Module._resolveFilename (module.js:351:15)
    at Function.Module._load (module.js:293:25)
    at Module.require (module.js:380:17)
    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/atom/browser/lib/rpc-server.js:128:79)
    at EventEmitter.emit (events.js:119:17)
    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/atom/browser/api/lib/web-contents.js:99:23)
    at EventEmitter.emit (events.js:119:17)

```
At /Applications/Atom.app/Contents/Resources/atom/renderer/api/lib/remote.js:77

Error: Cannot find module 'dialog'
Error: Cannot find module 'dialog'
    at Function.Module._resolveFilename (module.js:351:15)
    at Function.Module._load (module.js:293:25)
    at Module.require (module.js:380:17)
    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/atom/browser/lib/rpc-server.js:128:79)
    at EventEmitter.emit (events.js:119:17)
    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/atom/browser/api/lib/web-contents.js:99:23)
    at EventEmitter.emit (events.js:119:17)
    at metaToValue (/Applications/Atom.app/Contents/Resources/atom/renderer/api/lib/remote.js:77:15)
    at Object.exports.require (/Applications/Atom.app/Contents/Resources/atom/renderer/api/lib/remote.js:157:34)
    at Atom.module.exports.Atom.confirm (/Applications/Atom.app/Contents/Resources/app/src/atom.js:705:23)
    at TreeView.module.exports.TreeView.removeSelectedEntries (/Applications/Atom.app/Contents/Resources/app/node_modules/tree-view/lib/tree-view.js:895:19)
    at atom-workspace.disposables.add.atom.commands.add.tree-view:remove (/Applications/Atom.app/Contents/Resources/app/node_modules/tree-view/lib/main.js:84:39)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:243:29)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:156:19)
    at jQuery.fn.trigger (/Applications/Atom.app/Contents/Resources/app/src/space-pen-extensions.js:64:23)
    at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:80:67)
```

### Commands

```
  6x -4:21.1 core:move-up (atom-text-editor.editor)
     -4:19.1 core:move-left (atom-text-editor.editor)
     -4:15.5 core:save (atom-text-editor.editor)
     -2:19.6 core:backspace (atom-text-editor.editor)
  3x -2:17.4 core:move-right (atom-text-editor.editor)
  2x -2:07.9 editor:move-to-first-character-of-line (atom-text-editor.editor)
     -2:07.4 core:select-down (atom-text-editor.editor)
     -2:06.5 core:copy (atom-text-editor.editor)
     -2:04.1 core:move-up (atom-text-editor.editor)
     -2:03.9 core:move-down (atom-text-editor.editor)
     -2:03.7 core:select-down (atom-text-editor.editor)
     -2:03.4 core:copy (atom-text-editor.editor)
  2x -2:00.7 editor:move-to-first-character-of-line (atom-text-editor.editor)
  3x -2:00.3 core:select-down (atom-text-editor.editor)
     -1:59.8 core:paste (atom-text-editor.editor)
  3x -1:58.7 core:save (atom-text-editor.editor)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ],
    "destroyEmptyPanes": false
  },
  "tree-view": {}
}
```

### Installed Packages

```coffee
# User
language-haml, v0.15.0

# Dev
No dev packages
```

**Comment by 50Wliu** [comment]
@benogle what's with these super-long titles?

**Comment by izuzak** [comment]
Hmm, not sure why this might be happening -- that require shouldn't be failing:

https://github.com/atom/atom/blob/b05a36d9f0667d68d06fc02fff1620deed9bbb0a/src/atom.coffee#L680

@SaxxonPike Was this a one-time exception or is this consistently happening for you? 

Also cc @kevinsawicki for :eyes:

**Comment by benogle** [comment]
> @benogle what's with these super-long titles?

Maybe github is accepting more text for titles now? I should probably manually truncate the title.

**Comment by kevinsawicki** [comment]
@zcbenz any thoughts on this look? Looks like it is thrown by calling `require('remote').require('dialog')`

**Comment by zcbenz** [comment]
I have no idea why it happened, it looks like something went wrong in module search paths on the browser side.

**Comment by dasilvacontin** [comment]
I came up with this exception while closing an Unsaved File via Ctrl.+W.

**Comment by izuzak** [comment]
@kevinsawicki This looks very similar to https://github.com/atom/atom/issues/6548 -- was this one perhaps fixed by https://github.com/atom/atom/pull/7560 as well or does it need a :syringe: as well?

**Comment by kevinsawicki** [comment]
> This looks very similar to atom/atom#6548 -- was this one perhaps fixed by atom/atom#7560 as well or does it need a :syringe: as well?

Yeah, this is slightly different than the dialog is required via the remote module, not sure the same fix will help, it might be possible to require dialog in the browser process early on so it caches it and then this error might possibly go away.

**Comment by arhatlohan** [comment]
I have this problem too.
When I chose the Atom, and reopen it.
It works !
